job-application-tracker-api/src/applications/dto/update-application.dto.ts

```diff
@@ -0,0 +1,27 @@
+import { IsNotEmpty,IsString } from 'class-validator';
+import { ApiProperty } from '@nestjs/swagger';
+
+export class UpdateApplicationDto{
+    @IsString()
+    @IsNotEmpty()
+    @ApiProperty()
+    id: string;
+
+    @IsString()
+    @IsNotEmpty()
+    @ApiProperty()
+    companyName: string;
+
+    @IsString()
+    @IsNotEmpty()
+    @ApiProperty()
+    position: string;
+
+    @ApiProperty()
+    states: any[];
```

Comment:
No need to allow modifing the states array in the update request. We will have an API for states